python/pyiceberg/exceptions.py

```diff
@@ -16,12 +16,8 @@
 #  under the License.
 
 
-class NoSuchTableError(Exception):
```

Comment:
I don't think that we want to change these errors, since they are independent of the REST catalog. I'd leave the existing ones unchanged and just make the REST-specific errors extend `RESTError`

Reply:
I had the same concern, also others could be applicable outside of the REST contest. I'll move `NoSuchNamespaceError` and `NoSuchTableError` back to inherit from `Exception`.